From: Dieter Miosga  <dmiosga6200@gmail.com>
Date: Sun, 14 Jun 2015 23:59:59 -0000
Subject: [PATCH] Sorry, but 3dnowprefetch from FX-6100 understands 3dnow

According all my experience, a kernel compiled with -march=amdfam10 and -m3dnow 
does not crash on an AMD Bulldozer FX-series (bdver1) core 
with 3dnowprefetch but without 3dnow instruction set, and understands the 3dnow instructions!
Therefore, do not die but warn on missing kernel instructions for processor
and change the text message to the user.

Signed-off-by: Happy Lander <secure-piloting@freed-universe.com>
---
diff -uNr a/arch/x86/boot/cpu.c b/arch/x86/boot/cpu.c
--- a/arch/x86/boot/cpu.c	2014-12-07 22:21:05.000000000 +0000
+++ b/arch/x86/boot/cpu.c	2015-01-18 13:14:15.968113636 +0000
@@ -88,7 +88,7 @@
 	}
 
 	if (err_flags) {
-		puts("This kernel requires the following features "
-		     "not present on the CPU:\n");
+		puts("This kernel check detects requirement for following features "
+		     "not present on the running Processor(s):\n");
 		show_cap_strs(err_flags);
 		putchar('\n');
diff -uNr a/arch/x86/boot/main.c b/arch/x86/boot/main.c
--- a/arch/x86/boot/main.c	2014-12-07 22:21:05.000000000 +0000
+++ b/arch/x86/boot/main.c	2015-01-18 13:16:21.336110408 +0000
@@ -147,9 +147,14 @@
 
 	/* Make sure we have all the proper CPU support */
 	if (validate_cpu()) {
-		puts("Unable to boot - please use a kernel appropriate "
-		     "for your CPU.\n");
-		die();
+		puts("If you do not use compile configuration\n"
+		     "different from YaIniT,  this means:\n"
+		     "Unable to boot - \n"
+		     "please use a kernel appropriate for your CPU\n"
+		     "      ON ALL FUTURE FAILURES AND CRASHES !   ");
+		puts("Otherwise, using a configuration different from YaIniT,\n"
+		     "this kernel will crash on any use of the detected missing feature!");
+		/*die();*/
 	}
 
 	/* Tell the BIOS what CPU mode we intend to run in. */
